src/rmarkdown/knit.ts

```diff
@@ -149,6 +149,7 @@ export class RMarkdownKnitManager extends RMarkdownManager {
 
 	// alters the working directory for evaluating chunks
 	public setKnitDir(): void {
+		const currentDocumentWorkspace = vscode.workspace.getWorkspaceFolder(vscode.window.activeTextEditor.document.uri);
```

Comment:
If the document is untitled but set to `rmd` language, then the workspace folder is `undefined`, if I remember correctly.

Reply:
Ah, yes that would cause some trouble. Given that a document can't be knit without being saved, I think it would make sense to have the following fallbacks:

Workspace folder > document folder > NULL